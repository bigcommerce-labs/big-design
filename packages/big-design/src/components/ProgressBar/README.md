# Progress Bar

## Examples

<!-- STORY -->

<!-- SOURCE -->

---

### Determinant Progress

Determinant Progress is a known amount of time or completeness for a task to complete. To render a determinant progress bar, a `percent` property needs to be passed.

### Indeterminant Progress

Indeterminant Progress is an unknown amount of time for a task to complete. Will render an indeterminate progress bar if no `percent` property is passed.

## Properties & Methods

| Prop Name | Type     | Default | Description          |
| :-------- | :------- | :------ | :------------------- |
| `percent` | `number` |         | Sets the fill length |
