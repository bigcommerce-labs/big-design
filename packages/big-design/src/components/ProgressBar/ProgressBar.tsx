import React from 'react';

import { StyledProgressBar, StyledProgressBarFiller } from './styled';

export interface ProgressBarProps {
  percent?: number;
}

export class ProgressBar extends React.PureComponent<ProgressBarProps> {
  render() {
    const { percent } = this.props;

    return typeof percent === 'number' ? (
      <StyledProgressBar aria-valuemax={100} aria-valuemin={0} aria-valuenow={percent} role="progressbar">
        <StyledProgressBarFiller percent={percent} />
      </StyledProgressBar>
    ) : (
      <StyledProgressBar role="progressbar">
        <StyledProgressBarFiller />
      </StyledProgressBar>
    );
  }
}
