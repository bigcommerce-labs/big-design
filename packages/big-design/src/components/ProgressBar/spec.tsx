import 'jest-styled-components';
import React from 'react';
import { render } from 'react-testing-library';

import { ProgressBar } from './ProgressBar';

test('render determinant progress bar', () => {
  const { container } = render(<ProgressBar state="incomplete" percent={50} />);

  expect(container).toMatchSnapshot();
});

test('render indeterminant progress bar', () => {
  const { container } = render(<ProgressBar state="indeterminant" />);

  expect(container).toMatchSnapshot();
});
