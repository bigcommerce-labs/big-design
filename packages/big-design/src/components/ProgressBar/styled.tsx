import styled, { css, keyframes } from 'styled-components';

import { defaultTheme } from '../../theme';

import { ProgressBarProps } from './ProgressBar';

export const StyledProgressBarLinear = styled.div<ProgressBarProps>`
  background-color: ${({ theme }) => theme.colors.secondary20};
  height: 4px;
  width: 100%;
`;

StyledProgressBarLinear.defaultProps = { theme: defaultTheme };

export const StyledProgressBarCircular = styled.div<ProgressBarProps>``;

StyledProgressBarCircular.defaultProps = { theme: defaultTheme };

export const StyledFillerLinear = styled.div<ProgressBarProps>`
  background-color: ${({ theme }) => theme.colors.primary};
  height: 100%;

  ${({ behavior }) =>
    behavior === 'determinant'
      ? css<ProgressBarProps>`
          transition: width 0.2s ease-in;
          // fix this
          width: ${innerProps => `${innerProps.percent}%`};
        `
      : css`
          animation: ${slide} 3s linear infinite;
          position: relative;
          width: 10%;
        `};
`;

const slide = keyframes`
  0% {
    left: -10%;;
  }
  100% {
    left: 100%;
  }
`;

StyledFillerLinear.defaultProps = { theme: defaultTheme };

export const StyledFillerCircular = styled.div<ProgressBarProps>``;

StyledFillerCircular.defaultProps = { theme: defaultTheme };
