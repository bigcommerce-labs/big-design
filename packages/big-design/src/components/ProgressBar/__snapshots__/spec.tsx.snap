// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render determinant progress bar 1`] = `
.c0 {
  background-color: #ECEEF5;
  height: 0.25rem;
  overflow: hidden;
  width: 100%;
}

.c1 {
  background-color: #3C64F4;
  height: 100%;
  -webkit-transition: width 0.2s ease-in;
  transition: width 0.2s ease-in;
  width: 50%;
}

<div>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="50"
    class="c0"
    role="progressbar"
  >
    <div
      class="c1"
    />
  </div>
</div>
`;

exports[`render indeterminant progress bar 1`] = `
.c0 {
  background-color: #ECEEF5;
  height: 0.25rem;
  overflow: hidden;
  width: 100%;
}

.c1 {
  background-color: #3C64F4;
  height: 100%;
  -webkit-animation: iWofnF 3s ease-in-out infinite;
  animation: iWofnF 3s ease-in-out infinite;
  position: relative;
  width: 10%;
}

<div>
  <div
    class="c0"
    role="progressbar"
  >
    <div
      class="c1"
    />
  </div>
</div>
`;
